pkg/vsphere/disk/disk_manager.go

```diff
@@ -310,6 +315,9 @@ func (m *Manager) Detach(op trace.Operation, d *VirtualDisk) error {
 	}
 
 	if err := d.canBeDetached(); err != nil {
+		// even though canBeDetached() is called here and nowhere else, it does not imply
+		// an attempt to detach, so decrease the ref count from here for that reason
+		d.attachedRefs--
```

Comment:
Feels confused. If any error returned here, that means the disk cannot be detached, and no detach happens as well, why attachedRefs is reduced here?